### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by djc**
Would be nice to get this in a release.

Given the lack of activity, what will maintenance be like going forward?

### CODE REVIEW

**Comment by sebpuetz** [comment]
> Given the lack of activity, what will maintenance be like going forward?

I'm happy to review changes / dependency updates, I haven't been using the library (or rather anything in the project) for a couple of years so nothing has been driving active development.

I'll try to go through the other repos during the week and take care of some updates.

**Comment by djc** [comment]
Thanks for the quick response!

**Comment by djc** [comment]
Looks like the mips64 target is not available for Rust 1.72: https://github.com/rust-lang/rust/issues/115218.

Let me know how you want this handled?

**Comment by sebpuetz** [comment]
> Looks like the mips64 target is not available for Rust 1.72: [rust-lang/rust#115218](https://github.com/rust-lang/rust/issues/115218).
> 
> Let me know how you want this handled?

After a cursory look at the issue, it looks like RustCrypto is using PPC32 as an alternate BE target: https://github.com/RustCrypto/sponges/pull/62

I believe our intention was to test BE targets, too, so that should be a viable alternative

**Comment by djc** [comment]
I added a version bump to 0.18, since some of the dependencies that got bumped are visible in the public API.

**Comment by danieldk** [comment]
> I'm happy to review changes / dependency updates, I haven't been using the library (or rather anything in the project) for a couple of years so nothing has been driving active development.

Same here, I don't use finalfusion/finalfrontier anymore, but am happy to review stuf

File: src/chunks/metadata.rs [hunk 1]
```diff
@@ -16,18 +15,18 @@ use crate::io::ReadMetadata;
 /// finalfusion metadata in TOML format.
 #[derive(Clone, Debug, PartialEq)]
 pub struct Metadata {
-    inner: Value,
+    inner: toml::Table,
```
Comment (danieldk) [inline src/chunks/metadata.rs:18]: Any reason not to import `toml::Table` above?

File: src/compat/fasttext/io.rs [hunk 2]
```diff
@@ -107,7 +105,7 @@ impl ReadFastTextPrivate for Embeddings<FastTextSubwordVocab, NdArray> {
             )));
         }
 
-        let metadata = Value::try_from(config).map_err(|e| {
+        let metadata = toml::Table::try_from(config).map_err(|e| {
```
Comment (danieldk) [inline src/compat/fasttext/io.rs:108]: Same.

File: src/chunks/metadata.rs (hunk 1 above)
Comment (djc) [reply src/chunks/metadata.rs]: Changed it to import the type.

**Review by sebpuetz — APPROVED** [review]
LGTM

**Review by danieldk — APPROVED** [review]

**Comment by djc** [comment]
Thanks! Are there any other blockers to getting this published on crates.io?

**Comment by danieldk** [comment]
> Thanks! Are there any other blockers to getting this published on crates.io?

Nope, I'll get to it now.

**Comment by danieldk** [comment]
Done, thanks for making this happen!

**Pull request merged**